app/uk/gov/hmrc/tariffclassificationfrontend/controllers/CaseController.scala

```diff
@@ -86,15 +87,33 @@ class CaseController @Inject()(authenticatedAction: AuthenticatedAction,
     )
   }
 
-  def activityDetails(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
+  private def getCaseAndRenderView(reference: String, page: CaseDetailPage, toHtml: Case => Future[Html])
+                                  (implicit request: Request[_]): Future[Result] = {
+
+    casesService.getOne(reference).flatMap {
```

Comment:
For now, we can just remove the get case call, replace it with `request.case`

Reply:
@pllaurado - just checking - are you going to do this?   Can have the "caseExists" action on _all_ of the controller methods that deal with a single case.